test/built-ins/AsyncGeneratorFunction/instance-await-expr-in-param.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-asyncgeneratorfunction
+description: Definition of instance `length` property
+info: |
+    AsyncGeneratorFunction ( p1, p2, … , pn, body )
+    ...
+    3. Return CreateDynamicFunction(C, NewTarget, "async generator", args).
+
+    Runtime Semantics: CreateDynamicFunction
+    ...
+    29. If kind is "async" or "async generator", then
+        a. If parameters Contains AwaitExpression is true, throw a SyntaxError
+           exception.
+features: [async-iteration]
+flags: [async]
+---*/
+
+var AsyncGeneratorFunction = Object.getPrototypeOf(async function* () {}).constructor;
+
+// YieldExpression is permitted in function body.
+AsyncGeneratorFunction('x = await 42');
+
+assert.throws(SyntaxError, function() {
+  AsyncGeneratorFunction('x = await 42', '');
+}, 'AwaitExpression not permitted in parameters');
+
+var withinAsyncGenerator = async function*() {
+  AsyncGeneratorFunction('x = await 42', '');
+};
```

Comment:
this is tricky, but nice to have tested. +1

Reply:
I think you can credit to @jugglinmike for this tricky logic! He wrote the analogous test in GeneratorFunction directory (for yield) :)

Reply:
you could also add tests for the yield expressions here, maybe in a follow up PR?